mypy/test/data/check-inference.test

```diff
@@ -1415,13 +1415,16 @@ main: note: In member "f" of class "A":
 [case testAttributePartiallyInitializedToNoneWithMissingAnnotation]
 class A:
     def f(self) -> None:
-        self.x = None # E: Need type annotation for variable
+        self.x = None
```

Comment:
This should still produce `Need type annotation for variable`. Reasoning:

* Strict optional checking really shouldn't have an effect on this. This change seems unrelated to the rest of the PR.
* There's not enough information in the method scope to infer a non-`None` type for the attribute. I wouldn't like to infer the type of an attribute from two method bodies (type inference should be local for predictability).